packages/wallet-core/modules/wallet/operations.js

```diff
@@ -30,6 +30,10 @@ async function loadWalletTokens(wallet) {
 
     try {
       balance = await getTokenBalance(token.address, wallet.address);
+
+      if (balance === 'NaN') {
```

Comment:
`NaN` is a built in type , not string. The check for it is `isNaN()`

Anyway balance cannot be NaN, so getTokenBalance should throw an error instead of returning NaN